### PR TITLE
fix(core): Match order of pipeline config nav items to page sections

### PR DESCRIPTION
**Pull request by louisjimenez**
https://github.com/spinnaker/spinnaker/issues/3970

The feature flag check now occurs in the `triggers.directive` giving each `<page-section>` a chance to init and register its page in the correct order with the `PageNavigationState`. `PageNavigationState` can then be used to render navigation items as long as they are `visible`.

This includes `prettier` changes that I can split into a separate commit if helpful.

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/config/triggers/triggers.html [hunk 1]
```diff
@@ -4,50 +4,55 @@
...
-            <input type="checkbox" ng-model="pipeline.keepWaitingPipelines">
+            <input type="checkbox" ng-model="pipeline.keepWaitingPipelines" />
             <strong>Do not automatically cancel pipelines waiting in queue.</strong>
             <help-field key="pipeline.config.parallel.cancel.queue"></help-field>
           </label>
         </div>
       </div>
     </div>
   </page-section>
-  <render-if-feature feature="artifacts">
```
Comment (louisjimenez) [inline app/scripts/modules/core/src/pipeline/config/triggers/triggers.html]: Most of this file is prettier changes, but this page section is taken out of the `<render-if-feature>` and instead relies on `checkFeatureFlag`

File: app/scripts/modules/core/src/pipeline/config/triggers/triggers.html [hunk 2]
```diff
@@ -4,50 +4,55 @@
...
-          <button class="btn btn-block btn-add-trigger add-new" ng-click="triggersCtrl.addArtifact()">
-            <span class="glyphicon glyphicon-plus-sign"></span> Add Artifact
-          </button>
-        </div>
+  <page-section
+    key="artifacts"
+    label="Expected Artifacts"
+    badge="pipeline.expectedArtifacts.length"
+    no-wrapper="true"
+    visible="triggersCtrl.checkFeatureFlag('artifacts')"
```
Comment (louisjimenez) [inline app/scripts/modules/core/src/pipeline/config/triggers/triggers.html]: Calls `checkFeatureFlag` to determine visibility

**Comment by louisjimenez** [comment]
@sbwsg @maggieneterval fyi

**Comment by sbwsg** [comment]
@spinnakerbot cherry-pick 1.12

**Pull request merged**